File: OG.py
from BattleEngine import UnitAttributes, UnitKind

SmallCargo = UnitKind(0)
LargeCargo = UnitKind(1)
LightFighter = UnitKind(2)
HeavyFighter = UnitKind(3)
Cruiser = UnitKind(4)
Battleship = UnitKind(5)
ColonyShip = UnitKind(6)
Recycler = UnitKind(7)
EspionageProbe = UnitKind(8)
Bomber = UnitKind(9)
SolarSatellite = UnitKind(10)
Destroyer = UnitKind(11)
DeathStar = UnitKind(12)
Battlecruiser = UnitKind(13)
RocketLauncher = UnitKind(14)
LightLaser = UnitKind(15)
HeavyLaser = UnitKind(16)
GaussCannon = UnitKind(17)
IonCannon = UnitKind(18)
PlasmaTurret = UnitKind(19)
SmallShieldDome = UnitKind(20)
LargeShieldDome = UnitKind(21)

names = {
    SmallCargo: 'Small Cargo',
    LargeCargo: 'Large Cargo',
    LightFighter: 'Light Fighter',
    HeavyFighter: 'Heavy Fighter',
    Cruiser: 'Cruiser',
    Battleship: 'Battleship',
    ColonyShip: 'ColonyShip',
    Recycler: 'Recycler',
    EspionageProbe: 'Espionage Probe',
    Bomber: 'Bomber',
    SolarSatellite: 'Solar Satellite',
    Destroyer: 'Destroyer',
    DeathStar: 'Death Star',
    Battlecruiser: 'Battlecruiser',
    RocketLauncher: 'Rocket Launcher',
    LightLaser: 'Light Laser',
    HeavyLaser: 'Heavy Laser',
    GaussCannon: 'Gauss Cannon',
    IonCannon: 'Ion Cannon',
    PlasmaTurret: 'Plasma Turret',
    SmallShieldDome: 'Small Shield Dome',
    LargeShieldDome: 'Large Shield Dome',
}

units_attributes = {
    SmallCargo: UnitAttributes(
        weapons=5.0,
        shield=10.0,
        armor=4000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    LargeCargo: UnitAttributes(
        weapons=5.0,
        shield=25.0,
        armor=12000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    LightFighter: UnitAttributes(
        weapons=50.0,
        shield=10.0,
        armor=4000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    HeavyFighter: UnitAttributes(
        weapons=150.0,
        shield=25.0,
        armor=10000.0,
        rapid_fire={
            SmallCargo: 3,
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    Cruiser: UnitAttributes(
        weapons=400.0,
        shield=50.0,
        armor=27000.0,
        rapid_fire={
            LightFighter: 6,
            EspionageProbe: 5,
            SolarSatellite: 5,
            RocketLauncher: 10,
        },
    ),
    Battleship: UnitAttributes(
        weapons=1000.0,
        shield=200.0,
        armor=60000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    ColonyShip: UnitAttributes(
        weapons=50.0,
        shield=100.0,
        armor=30000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    Recycler: UnitAttributes(
        weapons=1.0,
        shield=10.0,
        armor=16000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    EspionageProbe: UnitAttributes(
        weapons=0.01,
        shield=0.01,
        armor=1000.0,
        rapid_fire={},
    ),
    Bomber: UnitAttributes(
        weapons=1000.0,
        shield=500.0,
        armor=75000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
            RocketLauncher: 20,
            LightLaser: 20,
            HeavyLaser: 10,
            GaussCannon: 5,
            IonCannon: 10,
            PlasmaTurret: 5,
        },
    ),
    SolarSatellite: UnitAttributes(
        weapons=1.0,
        shield=1.0,
        armor=2000.0,
        rapid_fire={},
    ),
    Destroyer: UnitAttributes(
        weapons=2000.0,
        shield=500.0,
        armor=110000.0,
        rapid_fire={
            EspionageProbe: 5,
            SolarSatellite: 5,
            Battlecruiser: 2,
            LightLaser: 10,
        },
    ),
    DeathStar: UnitAttributes(
        weapons=200000.0,
        shield=50000.0,
        armor=9000000.0,
        rapid_fire={
            SmallCargo: 250,
            LargeCargo: 250,
            LightFighter: 200,
            HeavyFighter: 100,
            Cruiser: 33,
            Battleship: 30,
            ColonyShip: 250,
            Recycler: 250,
            EspionageProbe: 1250,
            Bomber: 25,
            SolarSatellite: 1250,
            Destroyer: 5,
            Battlecruiser: 15,
            RocketLauncher: 200,
            LightLaser: 200,
            HeavyLaser: 100,
            GaussCannon: 50,
            IonCannon: 100,
        },
    ),
    Battlecruiser: UnitAttributes(
        weapons=700.0,
        shield=400.0,
        armor=70000.0,
        rapid_fire={
            SmallCargo: 3,
            LargeCargo: 3,
            HeavyFighter: 4,
            Cruiser: 4,
            Battleship: 7,
            EspionageProbe: 5,
            SolarSatellite: 5,
        },
    ),
    RocketLauncher: UnitAttributes(
        weapons=80.0,
        shield=10.0,
        armor=2000.0,
        rapid_fire={},
    ),
    LightLaser: UnitAttributes(
        weapons=100.0,
        shield=25.0,
        armor=2000.0,
        rapid_fire={},
    ),
    HeavyLaser: UnitAttributes(
        weapons=250.0,
        shield=100.0,
        armor=8000.0,
        rapid_fire={},
    ),
    GaussCannon: UnitAttributes(
        weapons=1100.0,
        shield=200.0,
        armor=35000.0,
        rapid_fire={},
    ),
    IonCannon: UnitAttributes(
        weapons=150.0,
        shield=500.0,
        armor=8000.0,
        rapid_fire={},
    ),
    PlasmaTurret: UnitAttributes(
        weapons=3000.0,
        shield=300.0,
        armor=100000.0,
        rapid_fire={},
    ),
    SmallShieldDome: UnitAttributes(
        weapons=1.0,
        shield=2000.0,
        armor=20000.0,
        rapid_fire={},
    ),
    LargeShieldDome: UnitAttributes(
        weapons=1.0,
        shield=10000.0,
        armor=100000.0,
        rapid_fire={},
    ),
}

File: OG.php
<?php

declare(strict_types=1);

namespace OG;

require_once 'BattleEngine.php';

use BattleEngine\UnitAttributes;

class OG
{
    public const SMALL_CARGO = 0;
    public const LARGE_CARGO = 1;
    public const LIGHT_FIGHTER = 2;
    public const HEAVY_FIGHTER = 3;
    public const CRUISER = 4;
    public const BATTLESHIP = 5;
    public const COLONY_SHIP = 6;
    public const RECYCLER = 7;
    public const ESPIONAGE_PROBE = 8;
    public const BOMBER = 9;
    public const SOLAR_SATELLITE = 10;
    public const DESTROYER = 11;
    public const DEATH_STAR = 12;
    public const BATTLECRUISER = 13;
    public const ROCKET_LAUNCHER = 14;
    public const LIGHT_LASER = 15;
    public const HEAVY_LASER = 16;
    public const GAUSS_CANNON = 17;
    public const ION_CANNON = 18;
    public const PLASMA_TURRET = 19;
    public const SMALL_SHIELD_DOME = 20;
    public const LARGE_SHIELD_DOME = 21;

    public static $unitsAttributes;

    public static $names = [
        self::SMALL_CARGO => 'Small Cargo',
        self::LARGE_CARGO => 'Large Cargo',
        self::LIGHT_FIGHTER => 'Light Fighter',
        self::HEAVY_FIGHTER => 'Heavy Fighter',
        self::CRUISER => 'Cruiser',
        self::BATTLESHIP => 'Battleship',
        self::COLONY_SHIP => 'Colony Ship',
        self::RECYCLER => 'Recycler',
        self::ESPIONAGE_PROBE => 'Espionage Probe',
        self::BOMBER => 'Bomber',
        self::SOLAR_SATELLITE => 'Solar Satellite',
        self::DESTROYER => 'Destroyer',
        self::DEATH_STAR => 'Death Star',
        self::BATTLECRUISER => 'Battlecruiser',
        self::ROCKET_LAUNCHER => 'Rocket Launcher',
        self::LIGHT_LASER => 'Light Laser',
        self::HEAVY_LASER => 'Heavy Laser',
        self::GAUSS_CANNON => 'Gauss Cannon',
        self::ION_CANNON => 'Ion Cannon',
        self::PLASMA_TURRET => 'Plasma Turret',
        self::SMALL_SHIELD_DOME => 'Small Shield Dome',
        self::LARGE_SHIELD_DOME => 'Large Shield Dome',
    ];
}

OG::$unitsAttributes = [
    OG::SMALL_CARGO => new UnitAttributes(
        5.0,
        10.0,
        4000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::LARGE_CARGO => new UnitAttributes(
        5.0,
        25.0,
        12000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::LIGHT_FIGHTER => new UnitAttributes(
        50.0,
        10.0,
        4000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::HEAVY_FIGHTER => new UnitAttributes(
        150.0,
        25.0,
        10000.0,
        [
            OG::SMALL_CARGO => 3,
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::CRUISER => new UnitAttributes(
        400.0,
        50.0,
        27000.0,
        [
            OG::LIGHT_FIGHTER => 6,
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
            OG::ROCKET_LAUNCHER => 10,
        ],
    ),
    OG::BATTLESHIP => new UnitAttributes(
        1000.0,
        200.0,
        60000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::COLONY_SHIP => new UnitAttributes(
        50.0,
        100.0,
        30000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::RECYCLER => new UnitAttributes(
        1.0,
        10.0,
        16000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::ESPIONAGE_PROBE => new UnitAttributes(
        0.01,
        0.01,
        1000.0,
        [],
    ),
    OG::BOMBER => new UnitAttributes(
        1000.0,
        500.0,
        75000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
            OG::ROCKET_LAUNCHER => 20,
            OG::LIGHT_LASER => 20,
            OG::HEAVY_LASER => 10,
            OG::GAUSS_CANNON => 5,
            OG::ION_CANNON => 10,
            OG::PLASMA_TURRET => 5,
        ],
    ),
    OG::SOLAR_SATELLITE => new UnitAttributes(
        1.0,
        1.0,
        2000.0,
        [],
    ),
    OG::DESTROYER => new UnitAttributes(
        2000.0,
        500.0,
        110000.0,
        [
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
            OG::BATTLECRUISER => 2,
            OG::LIGHT_LASER => 10,
        ],
    ),
    OG::DEATH_STAR => new UnitAttributes(
        200000.0,
        50000.0,
        9000000.0,
        [
            OG::SMALL_CARGO => 250,
            OG::LARGE_CARGO => 250,
            OG::LIGHT_FIGHTER => 200,
            OG::HEAVY_FIGHTER => 100,
            OG::CRUISER => 33,
            OG::BATTLESHIP => 30,
            OG::COLONY_SHIP => 250,
            OG::RECYCLER => 250,
            OG::ESPIONAGE_PROBE => 1250,
            OG::BOMBER => 25,
            OG::SOLAR_SATELLITE => 1250,
            OG::DESTROYER => 5,
            OG::BATTLECRUISER => 15,
            OG::ROCKET_LAUNCHER => 200,
            OG::LIGHT_LASER => 200,
            OG::HEAVY_LASER => 100,
            OG::GAUSS_CANNON => 50,
            OG::ION_CANNON => 100,
        ],
    ),
    OG::BATTLECRUISER => new UnitAttributes(
        700.0,
        400.0,
        70000.0,
        [
            OG::SMALL_CARGO => 3,
            OG::LARGE_CARGO => 3,
            OG::HEAVY_FIGHTER => 4,
            OG::CRUISER => 4,
            OG::BATTLESHIP => 7,
            OG::ESPIONAGE_PROBE => 5,
            OG::SOLAR_SATELLITE => 5,
        ],
    ),
    OG::ROCKET_LAUNCHER => new UnitAttributes(
        80.0,
        20.0,
        2000.0,
        [],
    ),
    OG::LIGHT_LASER => new UnitAttributes(
        100.0,
        25.0,
        2000.0,
        [],
    ),
    OG::HEAVY_LASER => new UnitAttributes(
        250.0,
        100.0,
        8000.0,
        [],
    ),
    OG::GAUSS_CANNON => new UnitAttributes(
        1100.0,
        200.0,
        35000.0,
        [],
    ),
    OG::ION_CANNON => new UnitAttributes(
        150.0,
        500.0,
        8000.0,
        [],
    ),
    OG::PLASMA_TURRET => new UnitAttributes(
        3000.0,
        300.0,
        100000.0,
        [],
    ),
    OG::SMALL_SHIELD_DOME => new UnitAttributes(
        1.0,
        2000.0,
        20000.0,
        [],
    ),
    OG::LARGE_SHIELD_DOME => new UnitAttributes(
        1.0,
        10000.0,
        100000.0,
        [],
    ),
];
